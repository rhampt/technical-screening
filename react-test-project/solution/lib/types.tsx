export enum TempUnit {
  CELSIUS = 'metric',
  FAHRENHEIT = 'imperial',
}

export interface IUSState {
  name: string
  abbr: string
}

export interface IWeatherDetails {
  id: number
  name: string
  description: string
  icon: string
  temp: number
  temp_min: number
  temp_max: number
  temp_feels: number
  wind: number
  humidity: number
  pressure: number
}
