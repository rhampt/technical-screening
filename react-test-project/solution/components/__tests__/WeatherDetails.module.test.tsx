import React from 'react'
import WeatherDetails from '../WeatherDetails.module'
import renderer from 'react-test-renderer'
import { IWeatherDetails, TempUnit } from '../../lib/types'

describe('WeatherDetails', () => {
  const weather: IWeatherDetails = {
    id: 4930956,
    name: 'Boston',
    description: 'Rain',
    icon: '10d',
    temp: 48,
    temp_min: 45,
    temp_max: 51,
    temp_feels: 45,
    wind: 3,
    humidity: 93,
    pressure: 1018,
  }
  it('should match the snapshot', () => {
    const tree = renderer.create(<WeatherDetails weather={weather} units={TempUnit.CELSIUS} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
