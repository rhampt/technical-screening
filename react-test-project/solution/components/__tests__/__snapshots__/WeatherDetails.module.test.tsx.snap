// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetails should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="topSection"
  >
    <div
      className="location"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root locationIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
      <span
        className="locationName"
      >
        Boston
      </span>
    </div>
  </div>
  <div
    className="iconSection"
  >
    <img
      className="icon"
      src="http://openweathermap.org/img/wn/10d@2x.png"
    />
    <div
      className="temp"
    >
      <span>
        48
        °
        C
      </span>
    </div>
    <div
      className="mainInfo"
    >
      <div>
        Rain
      </div>
      <div>
        <span>
          45
          °/
          51
          °
        </span>
      </div>
      <div>
        <div>
          Feels like: 
          <span>
            45
            °
            C
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="detailsSection"
  >
    <div
      className="innerContainer"
    >
      <div
        className="detail"
      >
        <div
          className="detailLabel"
        >
          Humidity:
        </div>
        <div
          className="detaiValue"
        >
          93
        </div>
      </div>
      <div
        className="detail"
      >
        <div
          className="detailLabel"
        >
          Pressure:
        </div>
        <div
          className="detaiValue"
        >
          1018
        </div>
      </div>
      <div
        className="detail"
      >
        <div
          className="detailLabel"
        >
          Wind:
        </div>
        <div
          className="detaiValue"
        >
          3
        </div>
      </div>
      <div
        className="detail"
      >
        <div
          className="detailLabel"
        >
          Temp Min:
        </div>
        <div
          className="detaiValue"
        >
          <span>
            45
            °
            C
          </span>
        </div>
      </div>
      <div
        className="detail"
      >
        <div
          className="detailLabel"
        >
          Temp Max:
        </div>
        <div
          className="detaiValue"
        >
          <span>
            51
            °
            C
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
